src/data/nat/factorial.lean

```diff
@@ -125,75 +135,216 @@ begin
+@[simp] lemma asc_fact_zero (n : ℕ) : n.asc_fact 0 = 1 := rfl
 
-@[simp] lemma zero_desc_fac (k : ℕ) : desc_fac 0 k = k! :=
+@[simp] lemma zero_asc_fact (k : ℕ) : asc_fact 0 k = k! :=
 begin
   induction k with t ht, refl,
-  unfold desc_fac, rw [ht, zero_add, nat.factorial_succ]
+  unfold asc_fact, rw [ht, zero_add, nat.factorial_succ],
 end
 
-lemma desc_fac_succ {n k : ℕ} : desc_fac n k.succ = (n + k + 1) * desc_fac n k := rfl
+lemma asc_fact_succ {n k : ℕ} : n.asc_fact k.succ = (n + k + 1) * n.asc_fact k := rfl
 
-lemma succ_desc_fac (n : ℕ) : ∀ k, (n + 1) * desc_fac n.succ k = (n + k + 1) * desc_fac n k
-| 0 := by rw [add_zero, desc_fac_zero, desc_fac_zero]
-| (k + 1) := by rw [desc_fac, mul_left_comm, succ_desc_fac, desc_fac, succ_add, ← add_assoc]
+lemma succ_asc_fact (n : ℕ) : ∀ k, (n + 1) * asc_fact n.succ k = (n + k + 1) * n.asc_fact k
+| 0 := by rw [add_zero, asc_fact_zero, asc_fact_zero]
+| (k + 1) := by rw [asc_fact, mul_left_comm, succ_asc_fact, asc_fact, succ_add, ← add_assoc]
 
-/-- `desc_fac n k = (n + k)! / n!`. However, this lemma states a reformulation to avoid ℕ-division.
-See `nat.desc_fac_eq_div` if you really need the version that uses ℕ-division. -/
-theorem factorial_mul_desc_fac (n : ℕ) : ∀ k, n! * desc_fac n k = (n + k)!
-| 0 := by rw [desc_fac, add_zero, mul_one]
-| (k + 1) := by rw [desc_fac_succ, mul_left_comm, factorial_mul_desc_fac, ← add_assoc, factorial]
+/-- `n.asc_fact k = (n + k)! / n!`. However, this lemma states a reformulation to avoid ℕ-division.
+See `nat.asc_fact_eq_div` if you really need the version that uses ℕ-division. -/
+theorem factorial_mul_asc_fact (n : ℕ) : ∀ k, n! * n.asc_fact k = (n + k)!
+| 0 := by rw [asc_fact, add_zero, mul_one]
+| (k + 1) := by rw [asc_fact_succ, mul_left_comm, factorial_mul_asc_fact, ← add_assoc, factorial]
 
-/-- Avoid in favour of `nat.factorial_mul_desc_fac` if you can. ℕ-division isn't worth it. -/
-lemma desc_fac_eq_div (n k : ℕ) : desc_fac n k = (n + k)! / n! :=
+/-- Avoid in favor of `nat.factorial_mul_asc_fact` if you can. ℕ-division isn't worth it. -/
+lemma asc_fact_eq_div (n k : ℕ) : n.asc_fact k = (n + k)! / n! :=
 begin
   apply mul_left_cancel' (factorial_ne_zero n),
-  rw factorial_mul_desc_fac,
+  rw factorial_mul_asc_fact,
   exact (nat.mul_div_cancel' $ factorial_dvd_factorial $ le.intro rfl).symm
 end
 
-lemma desc_fac_of_sub {n k : ℕ} (h : n < k) :
-  (k - n) * (k - n).desc_fac n = (k - (n + 1)).desc_fac (n + 1) :=
+lemma asc_fact_of_sub {n k : ℕ} (h : k < n) :
+  (n - k) * (n - k).asc_fact k = (n - (k + 1)).asc_fact (k + 1) :=
 begin
-  set t := k - n.succ with ht,
-  suffices h' : k - n = t.succ, by rw [←ht, h', succ_desc_fac, desc_fac_succ],
+  set t := n - k.succ with ht,
+  suffices h' : n - k = t.succ, by rw [←ht, h', succ_asc_fact, asc_fact_succ],
   rw [ht, succ_eq_add_one, ←sub_sub_assoc h (succ_pos _), succ_sub_one],
 end
 
-end desc_fac
+lemma pow_le_asc_fact (n : ℕ) : ∀ (k : ℕ), (n + 1)^k ≤ n.asc_fact k
```

Comment:
```suggestion
lemma succ_pow_le_asc_fact (n : ℕ) : ∀ (k : ℕ), (n + 1)^k ≤ n.asc_fact k
```
The current name suggests something more like n^k <= n.asc_fact k, but you proved a stronger result.